internal/controller/compute/user/user.go

```diff
@@ -0,0 +1,208 @@
+	service userapi.Client
+	log     logging.Logger
+}
+
+func (eu *externalUser) Observe(ctx context.Context, mg resource.Managed) (managed.ExternalObservation, error) {
+	cr, ok := mg.(*v1alpha1.User)
+	if !ok {
+		return managed.ExternalObservation{}, errors.New(errNotUser)
+	}
+
+	userID := meta.GetExternalName(cr)
+	if userID == "" {
+		return managed.ExternalObservation{}, nil
+	}
+
+	observed, resp, err := eu.service.GetUser(ctx, userID)
+	if resp.HttpNotFound() {
+		return managed.ExternalObservation{}, nil
+	}
+	if err != nil {
+		return managed.ExternalObservation{}, errors.Wrap(err, "failed to get user by id")
+	}
+
+	props := observed.GetProperties()
+	cr.Status.AtProvider.UserID = *observed.GetId()
+	cr.Status.AtProvider.S3CanonicalUserID = *props.GetS3CanonicalUserId()
+
+	cr.SetConditions(xpv1.Available())
+	return managed.ExternalObservation{
+		ResourceExists:    true,
+		ResourceUpToDate:  userapi.IsUserUpToDate(cr, observed),
+		ConnectionDetails: connectionDetails(cr, observed),
+	}, nil
+}
+
+func connectionDetails(cr *v1alpha1.User, observed ionosdk.User) managed.ConnectionDetails {
+	props := observed.GetProperties()
+	return managed.ConnectionDetails{
+		"email": []byte(*props.GetEmail()),
+		xpv1.ResourceCredentialsSecretPasswordKey: []byte(cr.Spec.ForProvider.Password),
+	}
+}
+
+func (eu *externalUser) Create(ctx context.Context, mg resource.Managed) (managed.ExternalCreation, error) {
+	cr, ok := mg.(*v1alpha1.User)
+	if !ok {
+		return managed.ExternalCreation{}, errors.Wrap(errors.New(errNotUser), "create error")
+	}
+	cr.SetConditions(xpv1.Creating())
+
+	nprops := ionosdk.NewUserPropertiesPost()
+	userapi.SetUserProperties(*cr, nprops)
+	observed, resp, err := eu.service.CreateUser(ctx, *ionosdk.NewUserPost(*nprops))
+	if err != nil {
+		werr := errors.Wrap(err, "failed to create user")
+		return managed.ExternalCreation{}, compute.AddAPIResponseInfo(resp, werr)
+	}
+
+	props := observed.GetProperties()
+	cr.Status.AtProvider.UserID = *observed.GetId()
+	cr.Status.AtProvider.S3CanonicalUserID = *props.GetS3CanonicalUserId()
+
+	meta.SetExternalName(cr, *observed.GetId())
+
+	conn := connectionDetails(cr, observed)
+	cr.Spec.ForProvider.Password = ""
+
+	return managed.ExternalCreation{ConnectionDetails: conn}, nil
+}
+
+func (eu *externalUser) Update(ctx context.Context, mg resource.Managed) (managed.ExternalUpdate, error) {
+	cr, ok := mg.(*v1alpha1.User)
+	if !ok {
+		return managed.ExternalUpdate{}, errors.Wrap(errors.New(errNotUser), "update error")
+	}
+
+	userID := cr.Status.AtProvider.UserID
+
+	props := ionosdk.NewUserPropertiesPut()
+	userapi.SetUserProperties(*cr, props)
+	observed, resp, err := eu.service.UpdateUser(ctx, userID, *ionosdk.NewUserPut(*props))
+	if err != nil {
+		werr := errors.Wrap(err, "failed to update user")
+		return managed.ExternalUpdate{}, compute.AddAPIResponseInfo(resp, werr)
+	}
+
+	if err = compute.WaitForRequest(ctx, eu.service.GetAPIClient(), resp); err != nil {
```

Comment:
do we need to have this wait for the Create method?

Reply:
Yes. The response code is 202. You can see that in our docs. [api.docs.ionos.com](https://api.ionos.com/docs/cloud/v6/)

Reply:
I see you added compute.WaitForRequest to the Create method, which was not there before. That what i was asking about.